packages/istanbul-reports/lib/clover/index.js

```diff
@@ -135,12 +135,12 @@ class CloverReport extends ReportBase {
                 count,
                 type: 'stmt'
             };
-            const branchDetail = branchByLine[k];
+            const branchDetail = branchDetails[k];
 
-            if (branchDetail) {
+            if (branchDetail && branchDetail.type === 'if') {
```

Comment:
Oh, thanks for point out. I'll dig it further later.

Reply:
I have done a simple search in source code and a testing (Check it in [my repo](https://github.com/KiritaniAyaka/istanbul-coverage-report/blob/main/test/index.test.ts#L27-L49)). Except `if ` and `switch`, there are three types of branches:

`default-arg`: 
The `states` always have one member and it presents the number of times that default arguments are applied.

`cond-expr`: (ternary operator)
The `states` always have two members. The first presents the number of times that condition was judged as `true`, and the second presents the number of times that condition was judged as `false`.

`binary-expr`:
The number of members is equal to the number of conditions. The xth member presents the number of times that xth condition was calculated. And it has nothing to do with the nest and calculating order.

---

For `default-arg`, I think it should be considered a `statement` according to [the standard](https://gist.github.com/aik099/8556655#file-clover-xsd-L117-L119) mentioned at #695 .

For `cond-expr`, it could be deal with the same as `if`.

I have no good idea to deal with `binary-expr`. Maybe we can consider it as covered when all conditions are calculated?